mailpoet/tests/DataFactories/Subscriber.php

```diff
@@ -97,6 +97,7 @@ public function withIsWooCommerceUser($isWooCustomer = true) {
    * @return $this
    */
   public function withSegments(array $segments) {
+    $this->segments = [];
```

Comment:
Good catch! This reminds me that we solved a similar issues caused by reusing the same instance of data factory by returning a clone in every `withXXX` method (e.g. https://github.com/mailpoet/mailpoet/blob/9f790efbf0f032fc32e264dc33450286b362976e/mailpoet/tests/DataFactories/WooCommerceProduct.php#L173-L179)

Maybe we could use the approach in all data factories. This note is not relevant to this PR since it would probably mean doing more changes in usages of factories. Just wanted to share the idea...

Reply:
I like this and I agree that we should have the same approach in all data factories. Should we maybe create a Jira ticket for it?

Reply:
I agree about creating the Jira. I [asked other devs about their opinion on slack](https://a8c.slack.com/archives/C01GH835U90/p1647852662288569). If we get an agreement on the proposed change I'll prepare the ticket.